api/src/main/java/com/codeforcommunity/rest/ApiRouter.java

```diff
@@ -68,6 +69,8 @@
 
   // token duration is 60 minutes in milliseconds
   private static final long TOKEN_DURATION = 3600000;
+  
+  private HashMap<String, String> resetTokenMap = new HashMap<String, String>();
```

Comment:
Wait since this is for a JWT, do we need to store state in the hashmap at all? Could we use the existing methods for validating and blacklisting used JWTs instead?

So in other words if we validate the JWT and it passes, we already know that this is a valid reset password request (as nobody has tampered with it and the only way you would feasibly put in valid JWT is if we sent it to you)

Once we use a token, we don't want that token to be used again so we should blacklist it.

This solution would be more in line with what JWTs are "supposed" to be: stateless.

However, we could just keep this implementation, which would make limiting one reset password attempt per user very easy! [See below]